.github/workflows/regression-report.yml

```diff
@@ -122,24 +163,219 @@ jobs:
+
+          # Remove .ci-temp/checkstyle/ from the beginning of FOUND_FILE
+          RELATIVE_PATH=${FOUND_FILE#.ci-temp/checkstyle/}
+
+          echo "Processed RELATIVE_PATH = $RELATIVE_PATH"
+
+          # Generate a unique name for the config file
+          CONFIG_FILE_NAME=$(basename "$FOUND_FILE" .java)_config.xml
+          CONFIG_FILE_PATH="$OUTPUT_DIR/$CONFIG_FILE_NAME"
+
+          # Generate a name for the projects file
+          PROJECT_FILE_PATH="$OUTPUT_DIR/projects.properties"
+
+          # Run the extractor
+            java -jar extractor-all-2024-08-21.jar "$BASE_PATH" \
+            --input-file "$RELATIVE_PATH" \
+            "$CONFIG_FILE_PATH" \
+            "$PROJECT_FILE_PATH"
+
+          # Set the config and project file paths
+          ./.ci/append-to-github-output.sh "config_link" "$CONFIG_FILE_PATH"
+          ./.ci/append-to-github-output.sh "projects_link" "$PROJECT_FILE_PATH"
+
+          # Print the contents of the files to the logs
+          echo "Contents of CONFIG_FILE_PATH:"
+          cat "$CONFIG_FILE_PATH"
+
+      - name: Upload Config and Project Files
+        uses: actions/upload-artifact@v3
+        with:
+          name: generated-config-and-project-file
+          path: |
+            ${{ steps.generate_config.outputs.config_link }}
+            ${{ steps.generate_config.outputs.projects_link }}
+
+  handle_configs_in_pr_description:
+    needs: [ parse_comment, checkout_and_cache ]
+    if: needs.parse_comment.outputs.mode == 'configs_in_pr_description'
+    runs-on: ubuntu-latest
+    outputs:
+      config_link: ${{ steps.parse_description.outputs.config_link }}
+      projects_link: ${{ steps.parse_description.outputs.projects_link }}
+    steps:
+      - name: Restore cached checkouts
+        uses: actions/cache@v4
+        with:
+          path: |
+            .
+            .ci-temp/checkstyle
+          key: ${{ runner.os }}-checkouts-${{ github.sha }}
+
+      - name: Getting PR description
+        env:
+          ISSUE_BODY: ${{ github.event.issue.body }}
+          PULL_REQUEST_URL: ${{ github.event.issue.pull_request.url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          mkdir -p .ci-temp
+          # convert windows line endings to unix in event text
+          echo "$ISSUE_BODY" > .ci-temp/windows.txt
+          tr -d '\15\32' < .ci-temp/windows.txt > .ci-temp/text
+
+          curl --fail-with-body -X GET "${PULL_REQUEST_URL}" \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: token $GITHUB_TOKEN" \
+              -o .ci-temp/info.json
+
+          jq --raw-output .head.ref .ci-temp/info.json > .ci-temp/branch
+          jq --raw-output .head.sha .ci-temp/info.json > .ci-temp/commit_sha
+
+      - name: Parsing content of PR description
+        id: parse_description
+        run: |
+          ./.ci/diff-report.sh parse-pr-description-text
+
   make_report:
+    needs: [
+      parse_comment,
```

Comment:
do we need `parse_comment` ?
I think you just need `handle_xxxxx`

Reply:
We do. Because, inside `make_report` requires
```
      - name: Generate report
        env:
          BRANCH: ${{ needs.parse_comment.outputs.branch }}
```

Reply:
time to split `make_report` into `download_configs` and `make_report`


`make_report` should have comment `Expected config files .ci-temp/*.xml, .ci-temp/project.properties`

Reply:
done

Reply:
if this is done, please resolve.